e2e/nomostest/new.go

```diff
@@ -337,6 +346,21 @@ func FreshTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 		nt.gitPrivateKeyPath = downloadSSHKey(nt)
 	}
 
+	if opts.InstallNotificationServer {
+		nt.NotificationServer = &NotificationServer{}
+		if err := nt.NotificationServer.install(nt); err != nil {
+			t.Fatalf("failed to install notification server: %v", err)
+		}
+		if err := nt.NotificationServer.portForward(nt); err != nil {
+			t.Fatalf("failed to port-forward notification server: %v", err)
+		}
+		t.Cleanup(func() {
+			if err := nt.NotificationServer.uninstall(nt); err != nil {
+				t.Errorf("failed to uninstall notification server: %v", err)
+			}
+		})
+	}
```

Comment:
Should we move the installation in `setupTestCase`?

The `InstallNotificationServer` option works when `--share--test-env` is `false`, but for shared GKE environment, it will install the notification server for all test cases. The port-forwarding will remain open for all tests, which might has the same issue as the metrics validation on autopilot clusters.

If we move it into `setupTestCase`, it will only install for the particular test with the `InstallNotificationServer` option.

If we have the code snippet in `FreshTestEnv`, you'll need

Reply:
Ah you're right the flag is enabled per-testcase which means the installation should be scoped to a testcase. Let me take another pass.

Reply:
updated to install on a per-testcase basis